solidity/contracts/TokenFactory.sol

```diff
@@ -0,0 +1,28 @@
+//SPDX-License-Identifier: Unlicense
+
+pragma solidity ^0.8.0;
+
+import '@openzeppelin/contracts/utils/Context.sol';
+import './ERC20WithData.sol';
+import './ERC721WithData.sol';
+
+contract TokenFactory is Context {
+    event TokenCreate(address indexed contract_address, string name, string symbol, bool is_fungible, bytes data);
```

Comment:
the naming convention used by openzeppelin seems to be either nones (Transfer, Approval), or passive verbs (Minted, Burned). plus I feel it'd be good to add `contract` to disambiguate with events on individual tokens. so suggest changing this to `TokenContractCreated` or `TokenContractCreation`?

Reply:
These are the event names from the OpenZeppelin ERC20/ERC721 contracts - they do seem to be all nouns:
```
Transfer
Approval
ApprovalForAll
```
I think `TokenCreation`, `TokenPoolCreation`, or `TokenContractCreation` are all suitable if we want to use a similar noun. Curious also how we should rename the [event on erc1155](https://github.com/hyperledger/firefly-tokens-erc1155/blob/e3def39a9d261d3192fab0d6b1add0571aba39f2/solidity/contracts/ERC1155MixedFungible.sol#L26) to align.

Reply:
`TokenPoolCreation` to cover both ERC20/721 (contract instances) and ERC1155 (internal state)?